index.js

```
@@ -145,62 +145,63 @@ function populateContainer (container) {
+	const { containerSelector = 'body', domain = 'FT.com' } = appInfo;
+	let surveyData;
+
+	const container = document.querySelector(`${containerSelector} .n-feedback__container`);
+	container.classList.remove('n-feedback--hidden');
+	populateContainer(container, domain);
+	const trigger = document.querySelector(`${containerSelector} .n-feedback__container .n-feedback__survey-trigger`);
+
+	const overlayId = `feedback-overlay-${containerSelector}`;
+	const feedbackOverlay = new Overlay(overlayId, {
+		html: `<div class="feedback-overlay__loader-wrapper"><div class="o-loading o-loading--dark o-loading--large"></div></div>`,
+		fullscreen: true,
+		class: 'feedback-overlay',
+		zindex: 1001,
+		customclose: '.n-feedback__survey__close-button'
+	});
 
-				// run Validation as soon as you display the first block
-				const firstBlock = document.querySelector('.n-feedback__survey-block', feedbackOverlay.content);
-				runValidation(firstBlock);
+	if (trigger) {
+		trigger.addEventListener('click', () => {
+			if (!surveyData) {
+				getSurveyData(surveyId).then( data => {
+					if ( !data || (data && data.length === 0) ) {
+						console.error('Bad survey data');
+					} else {
+						surveyData = data
+
+						try {
+							const html = surveyBuilder.buildSurvey(surveyData, surveyId);
+							// TODO: Validate the html
+							if (!feedbackOverlay.visible) {
+								feedbackOverlay.open();
+							}
+
+							feedbackOverlay.content.innerHTML = html;
+							setBehaviour(feedbackOverlay, surveyData, surveyId, appInfo);
+
+							// run Validation as soon as you display the first block
+							const firstBlock = document.querySelectorAll('.n-feedback__survey-block', feedbackOverlay.content)[0];
+							runValidation(firstBlock);
+						} catch( err ){
+							container.classList.add('n-feedback--hidden');
+							trigger.classList.add('n-feedback--hidden');
+							console.error('Error at building survey', err);
```

Same here, I think we shouldn't hide the error. Maybe it worth ask UX or @rowanbeentje how they are used to proceed for user communication.